controllers/admin/AdminProductsController.php

```diff
@@ -1904,7 +1904,9 @@ public function processUpdate()
                 $this->copyFromPost($object, $this->table);
                 $object->indexed = 0;
 
-                if (Shop::isFeatureActive() && Shop::getContext() != Shop::CONTEXT_SHOP) {
+                // disable to fix https://github.com/PrestaShop/PrestaShop/issues/31287
+                // probably all multishop_check related things can be removed?
+                if (false && Shop::isFeatureActive() && Shop::getContext() != Shop::CONTEXT_SHOP) {
                     $values = (array) Tools::getValue('multishop_check', []);
                     $values['state'] = Product::STATE_SAVED;
```

Comment:
```suggestion
                    $values['state'] = Product::STATE_SAVED;
                    $values['id_manufacturer'] = true;
                    $values['ean13'] = true;
                    $values['mpn'] = true;
                    $values['reference'] = true;
```

that is a correct fix as we need to pass the fields which have to be updated to the `$object->setFieldsToUpdate`

Reply:
it's not an ideal way, but the old page will be deleted in PrestaShop 9 anyway, so to make it work, we can use this solution :)

Reply:
@cruftex could you apply my suggestions? :)

> I would like to delete all this multishop_check related code, if it is not used any more, rather than add more band-aid.

This is too much and unsafe to do in a patch version. This is an old page that is already deprecated, so we can stick to fixing what's absolutely necessary.

Reply:
@kpodemski I do another PR against the 8.1.x branch with the conservative approach as you suggested.